src/config/resolve-config.js

```diff
@@ -74,11 +74,10 @@ function _resolveConfig(filePath, opts, sync) {
   const arr = [load, loadEditorConfig].map(l => l(filePath, opts.config));
 
   const unwrapAndMerge = ([result, editorConfigured]) => {
-    const merged = Object.assign(
-      {},
-      editorConfigured,
-      mergeOverrides(Object.assign({}, result), filePath)
-    );
+    const merged = {
+      ...editorConfigured,
+      ...mergeOverrides({ ...result }, filePath)
```

Comment:
```suggestion
      ...mergeOverrides(result, filePath)
```
`mergeOverrides` doesn't modify this argument.

Reply:
Done.